### PR TITLE
Implement xpytest

### PR DESCRIPTION
**Pull request by imos**

### CODE REVIEW

**Comment by imos** [comment]
pfnCI, test this please.

File: pkg/pytest/pytest.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	}
+	if len(p.Files) == 0 {
+		return nil, errors.New("Pytest.Files must not be empty")
+	}
+	args = append(args, p.Files...)
+
+	// Check deadline.
+	deadline := p.Deadline
+	if deadline <= 0 {
+		return nil, fmt.Errorf("Pytest.Deadline must be postiive value")
```
Comment (disktnk) [inline pkg/pytest/pytest.go]: positive?

File: pkg/resourcebuckets/resource_buckets.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package resourcebuckets
+
+import "sync"
+
+// ResourceBuckets manages resource capacities.  This enables worker threads to
+// use limited resources wihtout exceeding their capacities.
```
Comment (disktnk) [inline pkg/resourcebuckets/resource_buckets.go]: without?

File: pkg/resourcebuckets/resource_buckets.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	cond       *sync.Cond
+}
+
+// ResourceUsage represents a usage of resource.
+type ResourceUsage struct {
+	Index int
+	Usage int
+}
+
+// NewResourceBuckets createa a new ResourceBuckets with buckets, each of which
```
Comment (disktnk) [inline pkg/resourcebuckets/resource_buckets.go]: create(s) ?

**Review by disktnk — APPROVED** [review]
it's nitpick, Great Job!

File: pkg/pytest/pytest.go (hunk 1 above)
Comment (imos) [reply pkg/pytest/pytest.go]: Done.

File: pkg/resourcebuckets/resource_buckets.go (hunk 2 above)
Comment (imos) [reply pkg/resourcebuckets/resource_buckets.go]: Done.

File: pkg/resourcebuckets/resource_buckets.go (hunk 3 above)
Comment (imos) [reply pkg/resourcebuckets/resource_buckets.go]: Done.

**Comment by imos** [comment]
pfnCI, test this please.

**Pull request merged**